### PR TITLE
Updated ActionResult class to hold a Core::TaskEntity

### PR DESCRIPTION
**Pull request by antonovchtd**
As per @vladimir-moiseiev comment, ActionResult struct can now hold either a std::optional<Core::TaskID>, std::vector<Core::TaskEntity>, OR (NEW) Core::TaskEntity, when only a single Task DTO is being returned.

### CODE REVIEW

File: src/utilities/ActionResult.cpp [hunk 1]
```diff
@@ -20,6 +23,9 @@ ActionResult::~ActionResult() {
                 id->Core::TaskID::~TaskID();
             break;
         case kEntity:
+            entity.Core::TaskEntity::~TaskEntity();
```
Comment (vladimir-moiseiev) [inline src/utilities/ActionResult.cpp]: Never call destructor directly!

**Review by vladimir-moiseiev — CHANGES_REQUESTED** [review]

File: src/utilities/ActionResult.cpp (hunk 1 above)
Comment (antonovchtd) [reply src/utilities/ActionResult.cpp]: ok, I removed this line

**Review by IlliaPlaksa — APPROVED** [review]

**Review by KharchenkoMaks — APPROVED** [review]

File: src/ui/actions/GetTaskToShowLabelsAction.cpp [hunk 2]
```diff
@@ -18,6 +18,5 @@ ActionResult GetTaskToShowLabelsAction::execute(const std::shared_ptr<ModelInter
     }
 
     std::vector<Core::TaskEntity> tasks = model->getTasks(*id);
-    tasks.erase(tasks.begin() + 1, tasks.end());
-    return {ActionResult::Status::SUCCESS, tasks};
+    return {ActionResult::Status::SUCCESS, tasks[0]};
```
Comment (KharchenkoMaks) [inline src/ui/actions/GetTaskToShowLabelsAction.cpp]: maybe there is a need to check if model returned any tasks by this id?

**Review by MichaelKupriianov — APPROVED** [review]

**Review by AntonApraksin — APPROVED** [review]

File: src/ui/actions/GetTaskToShowLabelsAction.cpp (hunk 2 above)
Comment (antonovchtd) [reply src/ui/actions/GetTaskToShowLabelsAction.cpp]: should be OK, but fixed to make it explicit

**Review by vladimir-moiseiev — APPROVED** [review]

**Pull request merged**